src/omit.ts

```diff
@@ -28,15 +28,15 @@ export function omit<T extends object, K extends keyof T>(
  * @dataLast
  * @category Object
  */
-export function omit<K extends PropertyKey>(
+export function omit<T extends Record<PropertyKey, unknown>, K extends keyof T>(
```

Comment:
We don't need to limit T to Records, this is only needed if you want to maintain typing for the omitFoo factory that we don't want to maintain.

```suggestion
export function omit<T, K extends keyof T>(
```